usercommandmgmt/usercommandmgmt.py

```diff
@@ -169,7 +170,194 @@ async def cc_edit(self, ctx, command: str.lower, *, text: str = None):
+                                )
+                            if is_created:
+                                self.activeDb.save_to_db(
+                                    req_data[1],
+                                    req_data[0].message.author.id,
+                                    False,
+                                    req_data[0].message.guild.id,
+                                )
+
+                                MESSAGE_TEMPLATE = """command successfully created!: 
+                                You can now use >: {command}
+                                """
+
+                                can_use_alert = MESSAGE_TEMPLATE.format(
+                                    command=req_data[2]
+                                )
+                                await req_data[0].message.author.send(can_use_alert)
+
+                        elif payload.emoji.name == "❌":
+                            await req_data[0].message.author.send(
+                                "Sorry, your proposed command was deemed inappropriate by the moderators"
+                            )
+
+                        self.create_command_queue.pop(message)
+
+    # utility commands for users
+    @commands.command()
+    async def command_count(self, ctx):
+        """
+        gets the number of commands created by the caller of this command
+        """
+        currently_used = self.activeDb.get_user_comm_quantity(
+            ctx.message.author.id, ctx.message.guild.id
+        )
+        current_max = self.get_highest_user_comm_allowance(ctx.message.author)
+
+        MESSAGE_TEMPLATE = """You have assigned {currrent_amt} out of {max_amt} commands available to you"""
+
+        msg = MESSAGE_TEMPLATE.format(currrent_amt=currently_used, max_amt=current_max)
+
+        await ctx.send(msg)
+
+    # utility commands for moderators
+    @checks.mod_or_permissions(administrator=True)
+    @commands.command()
+    async def togglemoderation(self, ctx):
+        """
+        sets command_moderation to false if true, and vice versa
+        """
+        command_moderation = True if not command_moderation else False
+        ctx.send("""Command moderation has been set to {command_moderation}""")
+
+    @checks.mod_or_permissions(administrator=True)
+    @commands.command()
+    async def setmodchannel(self, ctx):
+        """
+        sets command_moderation to false if true, and vice versa
+        """
+        message = ctx.message.split(" ")
+        if len(message) == 2:
+            if (
+                self.find_channel_by_name(
+                    message[1],
+                )
+                != None
+            ):
+                mod_channel_name = message[1]
+                ctx.send("""Mod channel set to {mod_channel_name}""")
+            else:
+                ctx.send("No channel with that name was found on this server")
+        else:
+            ctx.send("you must provide a channel name to set")
```

Comment:
You'll want to use the built-in infra to help you make this easier. See the `cc_create` method you overrode earlier. It has other arguments that you can pass in and use. If you provide type hints, the bot will make sure that the arguments you have are valid before running the method. If they're not valid or missing, the corresponding help message will print.

Also you'll want to update the docstring here to something more appropriate, I haven't done it below.
```suggestion
    async def setmodchannel(self, ctx, channel: discord.TextChannel):
        """
        sets command_moderation to false if true, and vice versa
        """
            mod_channel_name = channel.name
            ctx.send("""Mod channel set to {mod_channel_name}""")
```